raft.go

```diff
@@ -1354,6 +1354,8 @@ func (r *Raft) processLogs(index uint64, futures map[uint64]*logFuture) {
 		applyBatch(batch)
 	}
 
+	r.persistCommitIndex(index)
+
```

Comment:
I don't think this is the right place for this call. I think we should call it in `storeLogs` just before the actual call to `StoreLogs` on the storage since that is the write we want our log stores to be able to add this state to disk in. Calling it here will not do anything until the next StoreLogs call at least in the design I proposed in the issue.